app/controllers/api/v0/shipments_controller.rb

```diff
@@ -44,14 +44,16 @@ def update
 
       def ready
         authorize! :read, Spree::Shipment
-        unless @shipment.ready?
-          if @shipment.can_ready?
-            @shipment.ready!
-          else
-            render(json: { error: I18n.t(:cannot_ready, scope: "spree.api.shipment") },
-                   status: :unprocessable_entity) && return
-          end
+
+        unless @shipment.ready? || @shipment.can_ready?
+          return render(
+            json: { error: I18n.t(:cannot_ready, scope: "spree.api.shipment") },
+            status: :unprocessable_entity
+          )
         end
+
+        @shipment.ready!
+
```

Comment:
I don't think that this is right. The error condition seems to be converted correctly:

```rb
!@shipment.ready? && !@shipment.can_ready?
!(@shipment.ready? || @shipment.can_ready?)
=> unless @shipment.ready? || @shipment.can_ready?
```

But then `@shipment.ready!` is executed in any case while it used to be executed only if it wasn't ready yet. Unfortunately, we need to repeat that check if we want a guard clause and not introduce a new method.
```suggestion
        unless @shipment.ready? || @shipment.can_ready?
          return render(
            json: { error: I18n.t(:cannot_ready, scope: "spree.api.shipment") },
            status: :unprocessable_entity
          )
        end

        @shipment.ready! unless @shipment.ready?

```

Reply:
That makes sense. I almost had it.